edb/pgsql/compiler/relgen.py

```diff
@@ -1253,37 +1253,39 @@ def process_set_as_membership_expr(
         else:
             is_array_unpack = False
 
-        with newctx.subrel() as _, _.newscope() as subctx:
-            dispatch.compile(right_arg, ctx=subctx)
-            pathctx.get_path_value_output(
-                subctx.rel, right_arg.path_id,
-                env=subctx.env)
+        # We compile `x [NOT] IN y` into essentially
+        # `[NOT] EXISTS (SELECT ... WHERE x = y)`.
```

Comment:
Hm, a quick test shows that Postgres generates the same {semi|anti}-join plan for both formulations.  This is not a guarantee that this'll happen in every query, but seems OK to me at the moment.